cmake/Modules/FindOpenMP.cmake

```diff
@@ -0,0 +1,610 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
```

Comment:
Any reasons for adding this in CUDA specific PR?

Reply:
There seems to be a bug for the supplied version of cmake 3.10, which parses -fopenmp to the nvcc compiler. There should be a link to a issue on cmake issue list in the comments.

Reply:
I also added it to the file didn't I?

Reply:
I thought that referred to the file itself. There wasn't a reason accompanying the link

Reply:
Added note in the file as well as removal when Cmake 3.11 is required.